serenity-bdd/serenity-core
# Escape strings in json summary report

**Summary of this PR**
Fields in the json summary report that could contain special characters (such as newline \n) are now escaped.  If not escaped, then the resulting json is invalid.

**Intended effect**
All strings (such as _description_ and _errorMessage_ fields for scenario outcomes) will now be escaped.  The example below shows the escaping of a new line in the description field:

```
"outcomes": [
  {
    "result": "failure",
    "description": "Ensure consent renewal is available when viewing the list of existing consents\nEnsure that a successful consent renewal lasts for 90 days",
    "errorMessage": "Assertion error: 1 expectation failed.Expected status code <200> but was <404>."
  }
]
```

**How should this be manually tested?**
Generate a report with a multi-line scenario description and verify that the description field is escaped, e.g.

```
 Scenario: Renew an existing consent
    Ensure consent renewal is available when viewing the list of existing consents
    Ensure that a successful consent renewal lasts for 90 days
    Given I am a customer user admin
      And I have an Open Banking account with active consent
    When I renew my consent
    Then the consent is renewed
```

**Side effects**
None.

**Documentation**
No changes required.

**Relevant tickets**
#1928

**Screenshots (if appropriate)**
N/A
